device-scanner/zed-enhancer/systemd-units/zed-enhancer.service

```diff
@@ -9,6 +9,8 @@ After=device-scanner.socket
 [Service]
 Restart=on-failure
 Environment=RUST_LOG=info
+# Reconfigure the daemon and rescan the directory for enabled ZEDLETs:
+ExecStartPre=/bin/bash -c '! systemctl --quiet is-active zfs-zed.service || systemctl kill --signal HUP zfs-zed.service'
```

Comment:
Hrrm, if this was not active, wouldn't we still have the same issue we do now?

Reply:
It's active if and only if it's running. If it's not running, it will read zedlets on start. It could also be uninstalled at all, if zfs is not used.

So this is simple: hup if running, otherwise do nothing.

Reply:
Well, it *must* be running because we have a `Requires` and `After` on it in this unit.

I wonder if we gain anything by having this check here... Is it ever possible to have a race?

Reply:
I don't think race is possible.
A typical case would be upgrading zed-enhancer which would need to kick zed itself.

Worse case: zed will reread zedlets shortly after start.

I guess just sending the signal will do, I was mistakenly assuming that zed is optional.